### ISSUE TITLE
Split up routes

### ISSUE THREAD

**Issue by anupvarghese** [kashishgupta1990/hapi-api#13]
I was looking at this repo for trying out some things in  Hapi. It was helpful to understand the architecture. I think you could clean up the code to bit more modular. 

For eg:- Routes, Controllers & Validations. I can do a clean up and push a PR sometime later.

**Comment by kashishgupta1990** [comment]
@anupvarghese - Sure, I am happy to hear that 👍

**Comment by anupvarghese** [comment]
@kashishgupta1990 I have created a new repo https://github.com/AskNemo/dosto and this would be a base structure I would be referencing.

**Comment by kashishgupta1990** [comment]
@anupvarghese - Yah, I have seen, that was good to have separate files for validations and routes but as I observe if we continuously splitting the files from route then we end up with too many files with no reusability for validation with different routes. My suggestion is not to split files until we have any cases for reuse same validation for different routes. 

HapiJs framework is more like to pass config object which has route URL, handler, validation all together in one place. If you want to have a separate controller and separate route files you can try express framework as well. That will suits your need directly with no big changes.

You should have a look on 

https://github.com/angular-fullstack/generator-angular-fullstack
Maybe this will fits in your requirement.

**Comment by anupvarghese** [comment]
@kashishgupta1990 I think, you didn't check the routes correctly, it allows us to re-use validations in a better way check [here](https://github.com/AskNemo/dosto/blob/develop/src/routes/hello.js#L8).
I think that answers the second comment also, split to files and re-use the code as much as you can. Keeping everything in one route file make it bulky and hard to test. Yes, I was using express extensively and I believe Hapi allows to reuse and modularise in a better way. 
Thanks for the feedback anyways 👍 💯

**Comment by kashishgupta1990** [comment]
@anupvarghese - I can see only one route [hello](https://github.com/AskNemo/dosto/blob/develop/src/routes/hello.js) and one validation file for that.  

To have more understanding, can you provide some example in the same repo where you have 2 to 5 routes which are using the same validation file. Because each route expect differently body structure in real life, so there would be different validation to validate your request JSON. I am curious to see how you solve that problem, it would be learning for me. 

Thanks

**Comment by anupvarghese** [comment]
@kashishgupta1990 Yup, I am gonna add more things in a fork of this repo soon. Will share you the details 👍
